packages/core/console/src/lib/suggestions.ts

```diff
@@ -71,7 +71,13 @@ export default class Suggestions {
   }
 
   private listenToEvents() {
-    this.events.on("deploy:contract:deployed", (contract: any) => {
+    this.embark.events.once("deployment:deployContracts:afterAll", () => {
+      this.embark.events.request("contracts:list", (_: any, contracts: Contract[]) => {
+        contracts.forEach((contract) => { this.contracts[contract.className] = contract; });
+      });
+    });
+
+    this.events.on("deployment:contract:deployed", ({contract}: {contract: Contract}) => {
```

Comment:
I think you can move this (the `on`) inside the `once`, so that way, we only check for new contracts after having them all (we don't set `this.contracts[contract.className]` twice)

Reply:
That `on` is supposed to make newly deployed, i.e. at runtime, contract names available in the suggestions list, but I may misunderstand what you're suggesting.

Reply:
I mean that `deployment:contract:deployed` is called for each contract deployed, be it on the first run or on changed contracts. So on the first run, all contracts are gonna get "caught" by the `on` and then, `after:all` is going to get called and we're going to get all the contracts a second time.

I'm just suggesting that we move the `on` inside the `once`, so that we track contracts only after getting the original ones. 

eg:
```
this.embark.events.once("deployment:deployContracts:afterAll", () => {
  this.embark.events.request("contracts:list", (_: any, contracts: Contract[]) => {
    contracts.forEach((contract) => { this.contracts[contract.className] = contract; });
  });

  this.events.on("deployment:contract:deployed", ({contract}: {contract: Contract}) => {
    this.contracts[contract.className] = contract;

    // reset backing variable so contracts suggestions can be re-built for next request
    this._suggestions = [];
  });
});
```

Reply:
Ah, right, that makes sense, thanks!

Reply:
Updated.